### PR TITLE
displaying tasks and conversation with author in post-data-detail-view

### PR DESCRIPTION
**Pull request by Angamanga**
This pull request makes the following changes:
- Displays tasks in data-detail-mode
- Displays conversations with author in data-detail-mode
- Refactors data-detail-directive

Testing checklist:
- [x] go to data
- [x] Select a post with tasks
- [x] Tasks should be visible (if correct permissions)
- [x] Select a post without tasks
- [x] Tasks should not be visible
- [x] Select a post with author-contact-details (for example a tweet)
- [x] Conversation with author-box should be visible

Fixes ushahidi/platform# .

Ping @ushahidi/platform

### CODE REVIEW

**Comment by rowasc** [comment]
on it

**Comment by crcommons** [comment]
you're one step ahead of me @rowasc! Was about to review and got distracted by our bad UX.

File: app/main/posts/detail/post-detail-data.directive.js [hunk 1]
```diff
@@ -58,112 +58,122 @@ function PostDetailDataController(
...
-            $scope.form_description = results[0].description;
-            $scope.form_color = results[0].color;
-            $scope.tags = results[3];
-            // Set page title to '{form.name} Details' if a post title isn't provided.
-            if (!$scope.post.title) {
-                $translate('post.type_details', { type: results[0].name }).then(function (title) {
-                    $scope.$emit('setPageTitle', title);
+                _.each(tasks, function (task) {
+                    // Set post task id
+                    // NOTE: This assumes that there is only one Post Task per Post
```
Comment (rowasc) [inline app/main/posts/detail/post-detail-data.directive.js]: but there can be many tasks in a post. I am not sure about this part.

File: app/main/posts/detail/post-detail-data.directive.js [hunk 2]
```diff
@@ -58,112 +58,122 @@ function PostDetailDataController(
...
-        $scope.$emit('setPageTitle', $scope.post.title);
-    } else {
-        $translate('post.post_details').then(function (title) {
-            $scope.title = title;
-            $scope.$emit('setPageTitle', title);
-        });
-    }
+                }, $scope.form_attributes);
+                // Make the first task visible
+                if (!_.isEmpty(tasks) && tasks.length > 1) {
```
Comment (rowasc) [inline app/main/posts/detail/post-detail-data.directive.js]: I don't think we need to check !_.isEmpty if we also check tasks.length> 1 ?

**Pull request merged**